GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -2260,7 +2260,7 @@ private void generateListOfChangesInSubmodulesChangesToolStripMenuItem_Click(obj
             foreach (var (path, name) in modules)
             {
                 string diff = Module.RunGitCmd(
-                     string.Format("diff --cached -z -- {0}", name));
+                     string.Format("diff --submodule=short --cached -z -- {0}", name));
```

Comment:
As vbay pointed out, this should be handled in the same way as other commit commands
See for instance
https://github.com/gitextensions/gitextensions/blob/5f2d0234ad81cd29009f3f56a7de06bd657eb36b/GitCommands/Git/GitModule.cs#L2295

This command was missing also in 2.x, this is not the only override that may be needed for diff.
So the bugfix should just be done in a different way.

Reply:
Right. He is wanting to pull into master. My pull request will have all runGitCmd and similar calls using GitArgumentBuilder. That way all argument handling for git ... calls will be uniform. I am not changing the calls for log and such that have different types of handling. Just how the arguments get to the commands.